### PR TITLE
Issue 3080 - agent-install.sh uses new types for agent files

### PR DESCRIPTION
**Pull request by linggao**
Signed-off-by: linggao <linggao@us.ibm.com>

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have tagged the reviewers in a comment below incase my pull request is ready for a review
- [x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

### CODE REVIEW

File: agent-install/agent-install.sh [hunk 1]
```diff
@@ -147,7 +153,7 @@ function now() {
 
 # Get cmd line args. Each arg corresponds to an env var. The arg value will be stored in ARG_<envvar>. Then the get_variable function will choose the highest precedence variable set.
 # Note: could not put this in a function, because getopts would only process the function args
-AGENT_VERBOSITY=3   # default until we get it from all of the possible places
+AGENT_VERBOSITY=5   # default until we get it from all of the possible places
```
Comment (Polber) [inline agent-install/agent-install.sh:156]: Was this changed for your testing, or is it defaulting to 5 now?

File: agent-install/agent-install.sh [hunk 2]
```diff
@@ -321,14 +330,50 @@ function get_variable() {
...
+
+        if [[ -n $part2 ]]; then
+            if [[ $part2 == /* ]]; then
+                input_file_path=$INPUT_FILE_PATH
+            else
+                # new path with version
+                input_file_path="css:${CSS_OBJ_AGENT_SOFTWARE_BASE}-${part2}"
+            fi
+        else
+            # 'css:' case - need to use AgentFileVersion to decide if we can get the latest csert
```
Comment (Polber) [inline agent-install/agent-install.sh:354]: "csert" -> "cert"

File: agent-install/agent-install.sh [hunk 3]
```diff
@@ -384,21 +429,159 @@ function get_anax_release_version() {
...
+    fi
+
+    if [[ -n $AGENT_CERT_FILE && -f $AGENT_CERT_FILE ]]; then
+        cert_flag="--cacert $AGENT_CERT_FILE"
+    elif [[ -f $PERMANENT_CERT_PATH ]]; then
+        # using cert from a previous install, see if that works
+        cert_flag="--cacert $PERMANENT_CERT_PATH"
+    else
+        cert_flag="--insecure"
+    fi
```
Comment (Polber) [inline agent-install/agent-install.sh:459]: Should there be logic to automatically retry with --insecure if the old cert does not work?

File: agent-install/agent-install.sh [hunk 4]
```diff
@@ -471,15 +670,68 @@ function download_anax_release_file() {
 }
 
 # If necessary, download the cfg file from CSS.
+# The input_file_path is either AGENT_CFG_FILE or INPUT_FILE_PATH, the second 
+# parameter tells which one it is, 1 for AGENT_CFG_FILE 2 for INPUT_FILE_PATH.
```
Comment (Polber) [inline agent-install/agent-install.sh:674]: It looks like you used 1 and 0 for the second parameter when you call the function, not 1 and 2

File: agent-install/agent-install.sh (hunk 1 above)
Comment (linggao) [reply agent-install/agent-install.sh:156]: Yes, I will switch it back to 3.

File: agent-install/agent-install.sh [hunk 5]
```diff
@@ -2668,7 +2918,9 @@ function loadClusterAgentImage() {
             log_fatal 1 "Can not specify both AGENT_K8S_IMAGE_TAR_FILE and -i (INPUT_FILE_PATH)"
         fi
         if [[ $INPUT_FILE_PATH == css:* ]]; then
-            download_css_file "$INPUT_FILE_PATH/$AGENT_K8S_IMAGE_TAR_FILE"
+            local input_path
+            get_input_file_css_path input_path
+            download_css_file "$input_path/$AGENT_K8S_IMAGE_TAR_FILE"
```
Comment (LiilyZhang) [inline agent-install/agent-install.sh:2923]: Should we add this logic for downloading cronjob image at line: 2997 (https://github.com/open-horizon/anax/blob/63fba0a7a144c0978bfa2f660b1b92d6a5204c38/agent-install/agent-install.sh#L2997)

Comment (linggao) [reply agent-install/agent-install.sh:2923]: yes, will do.

File: agent-install/agent-install.sh (hunk 3 above)
Comment (linggao) [reply agent-install/agent-install.sh:459]: Tried, got a lot of problems for getting the cfg file. Will leave it this way for now.

File: agent-install/agent-install.sh (hunk 4 above)
Comment (linggao) [reply agent-install/agent-install.sh:674]: fixed

**Review by Polber — APPROVED** [review]

**Review by LiilyZhang — APPROVED** [review]

**Pull request merged**